### PR TITLE
[RW-9950][risk=low] Switch to use RSTUDIO app type in Leo. 

### PR DESCRIPTION
**Pull request by yonghaoy**
after https://github.com/DataBiosphere/leonardo/pull/3338, Leonardo introduced a new app type RSTUDIO, instead of the custom one with `descriptorPath`

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

**Review by dmohs — APPROVED** [review]
This is net ~-30 lines of code. Have my enthusiastic approval.

File: api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java [hunk 1]
```diff
@@ -250,11 +250,10 @@ KubernetesRuntimeConfig toKubernetesRuntimeConfig(
   LeonardoKubernetesRuntimeConfig toLeonardoKubernetesRuntimeConfig(
       KubernetesRuntimeConfig kubernetesRuntimeConfig);
 
-  @ValueMapping(source = "RSTUDIO", target = "CUSTOM")
   LeonardoAppType toLeonardoAppType(AppType appType);
 
-  @ValueMapping(source = "CUSTOM", target = "RSTUDIO")
   @ValueMapping(source = "GALAXY", target = MappingConstants.NULL) // we don't support Galaxy
+  @ValueMapping(source = "CUSTOM", target = MappingConstants.NULL) // we don't support Galaxy
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java:256]: comment is not quite right :)

File: api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java [hunk 2]
```diff
@@ -250,11 +250,10 @@ KubernetesRuntimeConfig toKubernetesRuntimeConfig(
   LeonardoKubernetesRuntimeConfig toLeonardoKubernetesRuntimeConfig(
       KubernetesRuntimeConfig kubernetesRuntimeConfig);
 
-  @ValueMapping(source = "RSTUDIO", target = "CUSTOM")
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java:253]: hooray!

**Review by jmthibault79 — COMMENTED** [review]
Hmm.  Is the change live on Leo yet?
```
[Fri May 26 15:34:28 EDT 2023] SEVERE: org.pmiops.workbench.leonardo.LeonardoRetryHandler$LeonardoRetryPolicy logNoRetry - Exception calling Leonardo API with response: The request content was malformed:
DecodingFailure at .appType: Invalid app type RSTUDIO
org.pmiops.workbench.leonardo.ApiException: Bad Request
```

**Comment by yonghaoy** [comment]
> Hmm. Is the change live on Leo yet?
> 
> ```
> [Fri May 26 15:34:28 EDT 2023] SEVERE: org.pmiops.workbench.leonardo.LeonardoRetryHandler$LeonardoRetryPolicy logNoRetry - Exception calling Leonardo API with response: The request content was malformed:
> DecodingFailure at .appType: Invalid app type RSTUDIO
> org.pmiops.workbench.leonardo.ApiException: Bad Request
> ```

Not yet. That Leo Pr is not merged yet... But I think Liz is very close!

**Pull request merged**